src/Uno.UI/UI/Xaml/FrameworkElement.macOS.cs

```diff
@@ -24,16 +26,76 @@ public override bool NeedsLayout
 		{
 			set
 			{
+				if (ShouldInterceptInvalidate)
+				{
+					return;
+				}
+
 				if (!_inLayoutSubviews)
 				{
 					base.NeedsLayout = value;
 				}
 
-				if (ShouldInterceptInvalidate)
+				SetLayoutFlags(LayoutFlag.MeasureDirty | LayoutFlag.ArrangeDirty);
+
+				SetSuperviewNeedsLayout();
+			}
+		}
+
+
+		public override void Layout()
```

Comment:
You may want to put Layout after `OnInvalidateMeasure` to make the diff easier to read :)